k8s/deployments.go

```diff
@@ -90,6 +95,61 @@ func (c deployments) CreateWelcomeDeployment(app *models.App) error {
 	return err
 }
 
+func (c deployments) create(
+	w io.WriteCloser,
+	app *models.App,
+	file *runtime.File,
+	deploy *deploy,
+	storage helpers.Storage,
+	done chan<- struct{},
+) {
+	// create log context with uuid and app name
+	lc := log.WithField("app", app.Name).WithField("deployUUID", deploy.uuid)
+	lc.Info("starting deploy...")
+	defer close(done)
+
+	// parsing teresa.yaml and updating app
+	if err := parsingTeresaYaml(app, file.Data); err != nil {
```

Comment:
If parsing `teresa.yaml` fails, the deployment goroutine shouldn't be created in the first place. Maybe we should try and parse it before creating the deployment goroutine and pass the `file.Data` to the deployment goroutine.

Also, the name of the function that would parse the `teresa.yaml` would be something as `parseTeresaYaml` instead of `parsingTeresaYaml`.